app/main.py

```diff
@@ -1 +1,52 @@
-#  write your code here
+from functools import reduce
+
+
+class Animal:
+    def __init__(self,
+                 name: str,
+                 appetite: int,
+                 is_hungry: bool = True) -> None:
+        self.name = name
+        self.appetite = appetite
+        self.is_hungry = is_hungry
+
+    def print_name(self) -> None:
+        print(f"Hello, I'm {self.name}")
+
+    def feed(self) -> int:
+        if not self.appetite or not self.is_hungry:
+            return 0
+
+        food_points_to_eat = self.appetite
+
+        print(f"Eating {food_points_to_eat} food points...")
+
+        self.is_hungry = False
+        self.appetite -= food_points_to_eat
+
+        return food_points_to_eat
```

Comment:
`feed - should print Eating {appetite} food points..., set is_hungry to False and return number of eaten food points if animal is hungry. Otherwise, it should return 0 and print nothing.` -- your function has slightly different logic

Reply:
Sorry, do not understand why the function has a different logic
It seems right to me

Reply:
Sorry, do not understand
It seems right to me

Reply:
I meant, you don't need to change appetite and is_hungry for current animal, according to the task description. But if you think it's appropriate to add here, it's fine, then I have a different comment 😁 Try to avoid creating additional `food_points_to_eat` variable here